vendor/golang.org/x/sys/unix/zsyscall_openbsd_amd64.go

```diff
@@ -527,6 +527,14 @@ func ioctl(fd int, req uint, arg uintptr) (err error) {
 	return
 }
 
+func ioctlPtr(fd int, req uint, arg unsafe.Pointer) (err error) {
+	_, _, e1 := syscall_syscall(libc_ioctl_trampoline_addr, uintptr(fd), uintptr(req), uintptr(arg))
+	if e1 != 0 {
+		err = errnoErr(e1)
+	}
+	return
+}
+
 var libc_ioctl_trampoline_addr uintptr
 
 //go:cgo_import_dynamic libc_ioctl ioctl "libc.so"
```

Comment:
The code patch adds a new function `ioctlPtr` that takes an `unsafe.Pointer` argument instead of a regular `uintptr` argument like in the original `ioctl` function. This seems to be a reasonable enhancement to support more complex ioctl operations that require passing a pointer argument.

However, it's not clear from this snippet what other parts of the codebase are affected and whether this change is consistent with the overall design and architecture of the system.

Some additional information would be helpful to assess if any bug risks or improvement suggestions exist.